app/src/main/java/com/github/lookupgroup27/lookup/ui/profile/Profile.kt

```diff
@@ -63,44 +66,64 @@ fun ProfileScreen(navigationActions: NavigationActions, avatarViewModel: AvatarV
-                // Profile Icon
-                val avatarRes = selectedAvatar ?: R.drawable.default_profile_icon
-                Icon(
-                    painter = painterResource(id = avatarRes),
-                    contentDescription = "Profile Icon",
-                    modifier = Modifier.size(150.dp),
-                    tint =
-                        if (avatarRes == R.drawable.default_profile_icon) Color.White
-                        else Color.Unspecified)
+                // Profile Icon Box for FAB placement
+                Box(contentAlignment = Alignment.BottomEnd) {
+                  // Profile Icon
+                  val avatarRes = selectedAvatar ?: R.drawable.default_profile_icon
+                  Icon(
+                      painter = painterResource(id = avatarRes),
+                      contentDescription = stringResource(R.string.profile_icon_description),
+                      modifier = Modifier.size(150.dp).padding(bottom = 8.dp),
+                      tint =
+                          if (avatarRes == R.drawable.default_profile_icon) Color.White
+                          else Color.Unspecified)
 
-                Button(
-                    onClick = { navigationActions.navigateTo(Screen.AVATAR_SELECTION) },
-                    colors = ButtonDefaults.buttonColors(DarkPurple)
-                    // modifier = Modifier.align(Alignment.Center)
-                    ) {
-                      Text("Change Avatar")
-                    }
+                  // Show FAB when no avatar is selected
+                  if (selectedAvatar == null || selectedAvatar == R.drawable.default_profile_icon) {
+                    FloatingActionButton(
+                        onClick = { navigationActions.navigateTo(Screen.AVATAR_SELECTION) },
+                        containerColor = DarkPurple,
+                        modifier =
+                            Modifier.size(36.dp) // Smaller size for Instagram-like style
+                                .offset(x = (-8).dp, y = (-8).dp) // Adjust position slightly
+                        ) {
+                          Icon(
+                              imageVector = Icons.Filled.Add,
+                              contentDescription = stringResource(R.string.add_avatar),
+                              tint = Color.White)
+                        }
+                  }
+                }
+
+                if (selectedAvatar != null && selectedAvatar != R.drawable.default_profile_icon) {
+                  // Show "Change Avatar" button only if a custom avatar is selected
+                  Button(
+                      onClick = { navigationActions.navigateTo(Screen.AVATAR_SELECTION) },
+                      modifier = Modifier.padding(top = 16.dp),
+                      colors = ButtonDefaults.buttonColors(DarkPurple)) {
+                        Text(stringResource(R.string.change_avatar))
+                      }
+                }
```

Comment:
Wouldn't it be better to use an `if-else` condition here instead of two `if` statements? This would make the logic clearer and help avoid potential issues with overlapping conditions.

Reply:
Thank you for pointing out the lack of clarity. But due to the layout structure, specifically with the Box and Column, implementing an if-else statement for the conditional rendering of the FAB and the "Change Avatar" button would not work as intended. This is because the Box and Column require independent control over their child elements, and merging the logic into an if-else would result in layout inconsistencies.
Instead, I opted to refactor the conditional logic by extracting the FAB and button into their respective composable functions, ProfileFab and ChangeAvatarButton. I think this approach keeps the logic modular and enhances code readability.